backend/runner/connector.py

```
@@ -0,0 +1,57 @@
+    async def connect(self):
+        if websockets is None:
+            logger.error("websockets library is not installed. Runner cannot start.")
+            return
+
+        headers = {"Authorization": f"Bearer {self.token}"}
+        try:
+            self.connection = await websockets.connect(self.target_ws_url, extra_headers=headers)
+            self.is_connected = True
+            logger.info(f"Connected to Naruon Gateway at {self.target_ws_url}")
+            await self._listen_loop()
+        except Exception as e:
```

Ruff (0.15.13)

[warning] Do not catch blind exception: `Exception`

(BLE001)